### PR TITLE
Service account refinements and pipeline changes

### PR DESCRIPTION
**Pull request by rocky341**
Hi @t92549 
Small changes in this PR.
Service account has been modified slightly to include delete permissions. Have also made the created role linked to the name of the deployment. 
Pipeline has been modified to also do a lint check.

### CODE REVIEW

File: .github/workflows/CodeAnalysisAndClean.yml [hunk 1]
```diff
@@ -14,9 +14,34 @@ jobs:
 
       - name: Run build
         run: mvn clean install
+  run-ui-tests:
```
Comment (t92549) [inline .github/workflows/CodeAnalysisAndClean.yml:17]: Can't this part stay in the `ci.yaml`?

Comment (K-itKat20) [reply .github/workflows/CodeAnalysisAndClean.yml]: Renamed this to run-ui-lint and removed the ui tests from this yml. The ci.yml still has UI unit tests.

**Comment by rocky341** [comment]
@t92549 
Is this ready to merge or do you require any additional changes?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -38,3 +38,6 @@ jobs:
     - name: Run tests
       run:
         npm run test
+    - name: Lint code
+      run:
+        npm run lint-errors
```
Comment (t92549) [inline .github/workflows/ci.yaml:43]: Isn't this run in both pipelines now?

Comment (K-itKat20) [reply .github/workflows/ci.yaml:43]: For now yes, unless you'd prefer to only have it one?

**Comment by t92549** [comment]
Happy to approve, just wondering though if it is normal that the "Code scanning results" section says 2 analyses not found?

**Comment by rocky341** [comment]
> Happy to approve, just wondering though if it is normal that the "Code scanning results" section says 2 analyses not found?

@t92549 
I think for this PR yes that is correct. I think it is because the file in which semgrep is run from has changed names.

**Review by t92549 — APPROVED** [review]

**Pull request merged**